frontend/src/components/file-explorer/TreeNode.test.tsx

```diff
@@ -1,148 +1,121 @@
 import React from "react";
-import { act, render } from "@testing-library/react";
+import { act } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { renderWithProviders } from "test-utils";
 import TreeNode from "./TreeNode";
-import { WorkspaceFile } from "#/services/fileService";
-
-const onFileClick = vi.fn();
-
-const NODE: WorkspaceFile = {
-  name: "folder",
-  children: [
-    { name: "file.ts" },
-    { name: "folder2", children: [{ name: "file2.ts" }] },
-  ],
-};
+import { selectFile, listFiles } from "#/services/fileService";
+
+vi.mock("../../services/fileService", async () => ({
+  listFiles: vi.fn(async (path: string = "/") => {
+    if (path === "/") {
+      return Promise.resolve(["folder1/", "file1.ts"]);
+    }
+    if (path === "/folder1/") {
+      return Promise.resolve(["file2.ts"]);
+    }
+    return Promise.resolve([]);
+  }),
+  selectFile: vi.fn(async () => Promise.resolve({ code: "Hello world!" })),
+  uploadFile: vi.fn(),
+}));
 
 describe("TreeNode", () => {
   afterEach(() => {
     vi.resetAllMocks();
   });
 
   it("should render a file if property has no children", () => {
-    const { getByText } = render(
-      <TreeNode
-        node={NODE}
-        path={NODE.name}
-        onFileClick={onFileClick}
-        defaultOpen
-      />,
+    const { getByText } = renderWithProviders(
+      <TreeNode path="/file.ts" defaultOpen />,
     );
 
     expect(getByText("file.ts")).toBeInTheDocument();
   });
 
-  it("should render a folder if property has children", () => {
-    const { getByText } = render(
-      <TreeNode
-        node={NODE}
-        path={NODE.name}
-        onFileClick={onFileClick}
-        defaultOpen
-      />,
+  it("should render a folder if it's in a subdir", async () => {
+    const { getByText } = renderWithProviders(
+      <TreeNode path="/folder1/" defaultOpen />,
     );
+    expect(listFiles).toHaveBeenCalledWith("/folder1/");
 
-    expect(getByText("folder")).toBeInTheDocument();
-    expect(getByText("file.ts")).toBeInTheDocument();
+    expect(await getByText("folder1")).toBeInTheDocument();
+    expect(await getByText("file2.ts")).toBeInTheDocument();
   });
```

Comment:
I can't figure out why this is failing. I think it has to do with the fact that `listFiles` is being called async?

Reply:
Yes, you are right. 

`get*` queries look for elements in the DOM and throw an error if none are found, `query*` queries do the same except they return null if not found instead of throw (useful for when you want to test an elements absence), and there is a third one, `find*` queries that "wait" for the element to appear (useful for testing async values).

In your case, you should rewrite the assertion to 
```ts
expect(await findByText("file2.ts")).toBeInTheDocument();
```

Reply:
```
afterEach(() => {
  // Replace vi.resetAllMocks with clearAllMocks
  // vi.resetAllMocks will clear mock history and reset its implementation to an empty function (will return undefined).
  // vi.resetAllMocks();
  // vi.clearAllMocks will clear mock history, but not reset its implementation to the default one.
  vi.clearAllMocks();
 });

 it("should render a folder if it's in a subdir", async () => {
  const { getByText } = renderWithProviders(
   <TreeNode path="/folder1/" defaultOpen />,
  );
  expect(getByText("folder1")).toBeInTheDocument();

  expect(listFiles).toHaveBeenCalledWith("/folder1/");

  // file2.ts is asynchronous, thus wrapped in waitFor with a 1s (default timeout)
  await waitFor(() => {
   expect(getByText("file2.ts")).toBeInTheDocument();
  });
 });
```

Reply:
@jiangleo `findBy` queries are a combination of `waitFor` and `getBy` queries internally (https://testing-library.com/docs/dom-testing-library/api-async/#findby-queries)

Reply:
Thanks all for the help here! Finally got all the tests re-enabled